### PR TITLE
Generate ci-operator jobs reporting failures to Sentry

### PR DESCRIPTION
**Pull request by petr-muller**
Generate jobs that mount the Sentry DSN secret and pass it to ci-operator.

Temporarily contain a commit causing integration test to fail. I want to use the faling jobs to test that the generated jobs are correct - I will re-run it modified to include the ci-operator changes.

### CODE REVIEW

**Comment by petr-muller** [comment]
Looks everything works:
```
2019/05/10 21:03:20 Container test in pod integration failed, exit code 2, reason Error
2019/05/10 21:03:26 Submitted failure event to sentry (id=f6a2f442ea0848eaabe3081dd1824352)
```

**Comment by petr-muller** [comment]
Breaking changes are expected, we intentionally change generated jobs. https://github.com/openshift/release/pull/3751 is the change that needs to merge after this PR.

File: cmd/ci-operator-prowgen/main.go [hunk 1]
```diff
@@ -22,6 +22,7 @@ import (
 
 const (
 	prowJobLabelVariant = "ci-operator.openshift.io/variant"
+	sentryDsnName       = "sentry-dsn"
```
Comment (stevekuznetsov) [inline cmd/ci-operator-prowgen/main.go]: Can we call this out as a volume name?

File: cmd/ci-operator-prowgen/main.go [hunk 2]
```diff
@@ -102,13 +103,29 @@ func generatePodSpec(info *config.Info, target string, additionalArgs ...string)
...
+					fmt.Sprintf("--target=%s", target),
+					"--sentry-dsn-path=/etc/sentry-dsn/ci-operator",
+				}, additionalArgs...),
+				Env: []kubeapi.EnvVar{{Name: "CONFIG_SPEC", ValueFrom: &configMapKeyRef}},
 				Resources: kubeapi.ResourceRequirements{
 					Requests: kubeapi.ResourceList{"cpu": *resource.NewMilliQuantity(10, resource.DecimalSI)},
 				},
+				VolumeMounts: []kubeapi.VolumeMount{{
+					Name:      sentryDsnName,
+					MountPath: "/etc/sentry-dsn",
```
Comment (stevekuznetsov) [inline cmd/ci-operator-prowgen/main.go]: Why not share this with the flag string?

File: cmd/ci-operator-prowgen/main.go (hunk 1 above)
Comment (petr-muller) [reply cmd/ci-operator-prowgen/main.go]: Yeah. It served also as a secret name, so I made a separate constant for it.

File: cmd/ci-operator-prowgen/main.go (hunk 2 above)
Comment (petr-muller) [reply cmd/ci-operator-prowgen/main.go]: Yeah, I extracted everything to constants.

**Comment by petr-muller** [comment]
Feedback addressed.

**Pull request merged**